weave

```diff
@@ -532,21 +533,28 @@ dns_args() {
     DNS_ARGS="$DNS_ARG $DNS_SEARCH_ARG"
 }
 
-# Perform operation $1 on container ID $2 to local DNS database at address $3
+# Perform operation $1 on container ID $2 to local DNS database with addresses at $3
+# followed by an optional fully-qualified domain name
 # This function is only called where we know $2 is a valid container name
 tell_dns() {
     METHOD="$1"
     CONTAINER_ID="$2"
     shift 2
-
+    collect_cidr_args "$@"
+    shift $CIDR_COUNT
```

Comment:
We have already done all the CIDR collecting at the call sites, so doing it here again, just so we can skip over them and look for a fqdn, is horrible. I suggest you introduce a `tell_dns_fqdn OPERATION CONTAINER FQDN CIDR1 CIDR2 ...` function, and invoke that from `tell_dns` and the `put-dns` code.